### ISSUE TITLE
docinfo is ignored in version 2.x of the plugin

### ISSUE THREAD

**Issue by dcp65** [asciidoctor/asciidoctor-gradle-plugin#368]
Trying to upgrade to version 2.1.0 of the asciidoctor-gradle-plugin.

I use a `docinfo.html` to include custom CSS and an import to the `tocbot.js` JavaScript library for an expandable table of contents.

My config:

```
plugins {
    id 'org.asciidoctor.jvm.convert' version '2.1.0'
}

repositories { jcenter() }

asciidoctor {
    sourceDir = file("src/docs/asciidoc/")
    ant.touch(file:"${sourceDir}/index.adoc")
    sources {
        include 'index.adoc'
    }
     resources {
        from(sourceDir) {
            include ('images/**', 'assets/js/**')
        }
    }
    outputDir "${project.buildDir}"
    attributes = [
            'doctype'      : 'article',
            'toc'          : 'left',
            'sectanchors'   : 'true',
            'toc-title'    :  'My Document',
            'source-highlighter': 'highlightjs',
            'imagesdir'    : './images',
            'docinfo1'     : ''
    ]
}
```

Result:

My `docinfo.html` is ignored, so that the custom CSS and JavaScript import are missing from the generated HTML.

**Comment by mojavelinux** [comment]
This may or may not be related, but you should not be using docinfo1
anymore. It should be docinfo: 'shared' or docinfo: 'private'.

**Comment by ysb33r** [comment]
@mojavelinux This will still use asciidoctorj 1.6, so I would expect `docinfo1` to work.

**Comment by mojavelinux** [comment]
I'm not suggesting docinfo1 wouldn't work (just that maybe there is something else going on). But the point is, no matter what version of AsciidoctorJ you're using, you shouldn't be using docinfo1. It's been deprecated for years.

**Comment by ysb33r** [comment]
@dcp65 problem reproduced, but no understanding yet why it is happening. Currently investigating this,

**Comment by ysb33r** [comment]
Oh my, it is the old issue and it is `baseDir` again. Right, I am going to fix it it in this way:

- #371 Use `UNSAFE` by default.
- #369 Allow thecontrolling of how baseDir is set
- Add a note on README to use `baseDirFollowsSourceDir()` when using `docinfo` with `docbook` and `html` backends. 
- Add code that detects at runtime that `docinfo` exists, but that a baseDir strategy has not been set and suggests that the author explicitly adds a strategy.

**Comment by ysb33r** [comment]
Fixed and scheduled for 2.2.0.
